### PR TITLE
Implemented roles 

### PR DESCRIPTION
**Pull request by OlexVoit**
Steps:
added entity role and create table
added a table of user-roles and changed the user entity
inserted roles into the table
inserted into the users-roles table
Fixed the user repository and

### CODE REVIEW

**Comment by OlexVoit** [comment]
I wrote a bad comment. That's because I split the branches a little bit wrong.

**Review by HovorukhaBohdan — APPROVED** [review]
Looks good!

File: src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - changeSet:
+      id: insert-users-roles
+      author: voitenko
+      changes:
+        - insert:
+            tableName: users_roles
+            columns:
+              - column:
+                  name: role_id
+                  value: "1"
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml]: Avoid hardcoding IDs, use valueComputed instead and select ID by email.

File: src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        - insert:
+            tableName: users_roles
+            columns:
+              - column:
+                  name: role_id
+                  value: "2"
+              - column:
+                  name: user_id
+                  value: "2"
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml]: Don't forget about empty lines.

File: src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      changes:
+        - createTable:
+            tableName: users_roles
+            columns:
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+                    foreignKeyName: role_foreignKey
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml]: Try to name your constraints globally unique. Here are the conventions: https://repository.prace-ri.eu/git/help/development/database/constraint_naming_convention.md

File: src/main/java/mate/academy/bookshop/config/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        auth -> auth
+                                .requestMatchers("/api/auth/**", "/error", "/swagger-ui/**")
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookshop/config/SecurityConfig.java]: Add also `/v3/api-docs/**` (it's a path to generated swagger docs)

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

File: src/main/java/mate/academy/bookshop/controller/AuthenticationController.java [hunk 5]
```diff
@@ -24,8 +26,13 @@ public class AuthenticationController {
...
-    UserResponseDto registerUser(@RequestBody UserRegistrationRequestDto requestDto)
+    UserResponseDto registerUser(@RequestBody @Valid UserRegistrationRequestDto requestDto)
             throws RegistrationException {
         return userService.register(requestDto);
     }
+
+    @PostMapping("/login")
+    boolean login(@RequestBody UserLoginRequestDto requestDto) {
+        return true;
+    }
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/controller/AuthenticationController.java]: You don't need to implement /login endpoint in this homework! remove this from pr.

File: src/main/java/mate/academy/bookshop/dto/user/UserLoginRequestDto.java
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/dto/user/UserLoginRequestDto.java:1]: remove not related to this hw changes from pr

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 6]
```diff
@@ -2,33 +2,92 @@
...
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/User.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 7]
```diff
@@ -2,33 +2,92 @@
...
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/User.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 8]
```diff
@@ -2,33 +2,92 @@
...
+    }
+
+    @Override
+    public String getUsername() {
+        return email;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/User.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/bookshop/service/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -24,6 +26,7 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throw new RegistrationException(CAN_NOT_REGISTER_USER_BY_EMAIL + requestDto.getEmail());
         }
         User userSave = userMapper.toModel(requestDto);
+        userSave.setPassword(passwordEncoder.encode(requestDto.getPassword()));
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/service/impl/UserServiceImpl.java:34]: what about role of new User?

File: src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml [hunk 10]
```diff
@@ -0,0 +1,26 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-users-roles-table
+      author: Oleksand voitenko
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml]:
```suggestion
      author: voitenko
```
everywhere you use

File: src/main/resources/db/changelog/changes/06-insert-user.yaml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+                  value: $2a$10$LmlI8w/AjPQt.OO1PwCAoOzax3GSPQ36zjWWFu.HhG/C1/hDbFxS2
+              - column:
+                  name: first_name
+                  value: Emma
+              - column:
+                  name: last_name
+                  value: Dumna
+              - column:
+                  name: shipping_address
+                  value: LA, Main Street.
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/06-insert-user.yaml:41]:
```suggestion
                  value: LA, Main Street.
                  
```

File: src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml [hunk 12]
```diff
@@ -0,0 +1,23 @@
+databaseChangeLog:
+  - changeSet:
+      id: insert-users-roles
+      author: Oleksand voitenko
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml]:
```suggestion
      author: voitenko
```
everywhere you use

File: src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml
Comment (d1sam) [inline src/main/resources/db/changelog/changes/07-insert-user-role-to-user-roles-table.yaml:1]: what about `preConditions`? you should use it together with `valueComputed`!

**Review by d1sam — CHANGES_REQUESTED** [review]
There are still some changes need to be done!

File: src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+      author: voitenko
+      changes:
+        - createTable:
+            tableName: users_roles
+            columns:
+              - column:
+                  name: role_id
+                  type: bigint
+                  constraints:
+                    nullable: false
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml]: you have primary key, that's why nullable constraint is unnecessary
```suggestion
```

File: src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+                    foreignKeyName: fk_users_roles_role
+                    references: roles(id)
+                    deleteCascade: true
+                    primaryKey: true
+              - column:
+                  name: user_id
+                  type: bigint
+                  constraints:
+                    nullable: false
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/04-create-user-roles-table.yaml]: you have primary key, that's why nullable constraint is unnecessary
```suggestion
```

**Review by d1sam — APPROVED** [review]
well done, but please look at my comments:)

**Pull request merged**